src/design-pages/passkey_authentication.rst

```diff
@@ -0,0 +1,318 @@
+centralized identity management system. For that purpose an integration with an
+LDAP server like AD or FreeIPA is needed.
+
+First, the key is registered in the LDAP server. Then, the SSSD clients should
+get the user credentials from this server, including the user key mapping data.
+On top of that, SSSD should communicate with the key to get the information
+stored in the device. Once this is done, SSSD should use all this data to
+authenticate the user.
+
+The biggest advantage of this approach is the usage of a central repository to
+store the user's credentials. In the long term this gives more flexibility as
+the users and keys can be authorized and revoked by the system administrator
+easily.
+
+Existing solution comparison
+****************************
+Existing Yubikey devices have already been enabled in RHEL and Fedora but the
+approach is completely different. The existing solution uses
+`OATH-TOTP <https://en.wikipedia.org/wiki/Time-based_one-time_password>`__ and
+`OATH-HOTP <https://en.wikipedia.org/wiki/HMAC-based_one-time_password>`__ to
+generate a one-time password (OTP) that is sent to the server as a second
+factor authentication. In other words, the authentication checks happen on the
+server. Moreover, this solution relies on a plugin for the 389 directory
+server, so it only works with IPA for IPA users.
+
+The approach explained in this document relies on the 
+`FIDO2 <https://en.wikipedia.org/wiki/FIDO2_Project>`__ standard, which uses a
+public-private key scheme to perform the authentication checks on the client.
+Besides, the solution is general enough to accept different LDAP servers like
+AD and IPA.
+
+Another difference is that with the current IPA feature, the first and second
+factor can also be used for doing a direct LDAP bind. This will not be possible
+with the new passkey scheme. Direct LDAP bind support will not be in the scope
+of this implementation.
+
+Implementation details
+----------------------
+
+The implementation will loosely follow existing smartcard authentication in
+SSSD. On top of that, and as mentioned before, the information will be stored
+in two places: in the LDAP server and on the passkey. The server will contain
+the public user credentials, including the user key mapping data, while the
+passkey device will store the private key necessary to authenticate the user.
+
+Key registration
+****************
+The first step is to register the user with a key. SSSD will provide a helper
+utility to handle the passkeys, and it can be used to create the key mapping
+string (similar to pamu2fcfg). Instead of calling the helper utility from the
+command line the system administrator would use 
+`sssctl <https://docs.pagure.org/sssd.sssd/design_pages/sssctl.html>`__. This
+would enable to unify the user experience. Then, the output of this command can
+be set to the LDAP server in a dedicated attribute. For AD we could use the
```

Comment:
is the attribute choice decided already for each ID provider? If so, please replace 'we could use' by a more definitive statement that lists the attributes per each ID provider.